packages/mui-docs/src/NProgressBar/NProgressBar.js

```diff
@@ -94,7 +38,52 @@ function NProgressBar(props) {
   return (
     <NoSsr>
       {props.children}
-      <GlobalStyles />
+      <GlobalStyles
+        styles={(theme) => ({
+          '#nprogress': {
+            direction: 'ltr',
+            pointerEvents: 'none',
+            position: 'fixed',
+            top: 0,
+            left: 0,
+            right: 0,
+            height: 2,
+            zIndex: (theme.vars || theme).zIndex.tooltip,
+            backgroundColor: (theme.vars || theme).palette.primary.light,
+            ...theme.applyDarkStyles({
+              backgroundColor: (theme.vars || theme).palette.primary.dark,
+            }),
+            '& .nprogress-bar': {
+              position: 'fixed',
+              backgroundColor: (theme.vars || theme).palette.primary.main,
```

Comment:
The store uses a different value. https://github.com/mui/mui-store/blob/3933fb75c13040c53dc16f4ea4286f39e03b63ca/web/src/utils/theme.js#L125

How can developers configure it?

For the backstory, the component was created with the objective to expose it like https://mantine.dev/others/nprogress/. I think that it would be great to have it in the lab.

Reply:
> The store uses a different value. https://github.com/mui/mui-store/blob/3933fb75c13040c53dc16f4ea4286f39e03b63ca/web/src/utils/theme.js#L125

It was `theme.palette.primary.main` before migrating to emotion. I think it should be the same.

> How can developers configure it?
> For the backstory, the component was created with the objective to expose it like https://mantine.dev/others/nprogress/. I think that it would be great to have it in the lab.

I think this is out of the scope of this PR but I do agree that it'd be useful to have it in the lab. I suggest to create a separate issue to keep track of it.

Reply:
@siriwatknp Actually, the store [doesn't use](https://github.com/mui/mui-store/search?q=NProgress.start%28%29%3B) NProgress.

> I think this is out of the scope of this PR

Agree, but I think that we should get clarity on the path forward, so we can align the current solution with it. In this case, maybe the right approach would be:

1. Migrate from `@mui/styles` to `@mui/system` (this PR)
2. Expose the component in the lab 
2. Remove the dependency on `nprogress`
3. Use `styles` rather than `GlobalStyles` so that developers can more easily use the sx prop to customize the component (unlocked by removing the dependency on `nprogress`).

Reply:
So, I think that we are good for customization. What I think we should fix though is the contract, I don't think that it's enough between light <-> main

Before:

https://user-images.githubusercontent.com/3165635/219010508-43551cac-04cc-408d-a877-28ffd2f67f6a.mov

After:

https://user-images.githubusercontent.com/3165635/219010903-56d07186-f175-421b-8bf5-c8280006a1fd.mov

Reply:
@oliviertassinari Fixed by using the same color as before.